soh/src/code/z_actor.c

```diff
@@ -1973,12 +1975,44 @@ s32 GiveItemWithoutActor(GlobalContext* globalCtx, s32 getItemId) {
     return false;
 }
 
+s32 GiveItemEntryFromActor(Actor* actor, GlobalContext* globalCtx, GetItemEntry getItemEntry, f32 xzRange, f32 yRange) {
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (!(player->stateFlags1 & 0x3C7080) && Player_GetExplosiveHeld(player) < 0) {
+        if ((((player->heldActor != NULL) || (actor == player->targetActor)) && 
+            ((!gSaveContext.n64ddFlag && ((getItemEntry.getItemId > GI_NONE) && (getItemEntry.getItemId < GI_MAX))) || 
+                (gSaveContext.n64ddFlag && ((getItemEntry.getItemId > RG_NONE) && (getItemEntry.getItemId < RG_MAX))))) ||
+                    (!(player->stateFlags1 & 0x20000800))) {
+            if ((actor->xzDistToPlayer < xzRange) && (fabsf(actor->yDistToPlayer) < yRange)) {
+                s16 yawDiff = actor->yawTowardsPlayer - player->actor.shape.rot.y;
+                s32 absYawDiff = ABS(yawDiff);
+
+                if ((getItemEntry.getItemId != GI_NONE) || (player->getItemDirection < absYawDiff)) {
+                    player->getItemEntry = getItemEntry;
+                    player->getItemId = getItemEntry.getItemId;
+                    player->interactRangeActor = actor;
+                    player->getItemDirection = absYawDiff;
+                    return true;
+                }
+            }
+        }
+    }
+
+    return false;
+}
+
+void GiveItemEntryFromActorWithFixedRange(Actor* actor, GlobalContext* globalCtx, GetItemEntry getItemEntry) {
+    GiveItemEntryFromActor(actor, globalCtx, getItemEntry, 50.0f, 10.0f);
+}
+
+// TODO: Rename to GiveItemIdFromActor or similar
```

Comment:
just a note, as of writing, it is still named `func_8002F434` in https://github.com/zeldaret/oot/blob/master/src/code/z_actor.c

Reply:
Yes, that's part of why we didn't go ahead and change it. This comment just being here improves search-ability and helps clear up what that function does even without the rename. Also since decomp moves kinda slow I thought eventually we might want to document some things ourselves down the line.